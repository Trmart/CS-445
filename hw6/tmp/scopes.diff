1,6c1,6
< .   .   Child: 0  Var: a of type int [mem: Local loc: -2 size: 1] [line: 3]
< .   .   .   Child: 0  Var: c of type char [mem: Local loc: -4 size: 1] [line: 6]
< .   .   .   Child: 0  Var: e of type bool [mem: Local loc: -4 size: 1] [line: 10]
< .   Child: 1  Compound [mem: None loc: 0 size: -4] [line: 2]
< .   .   Child: 1  Compound [mem: None loc: 0 size: -6] [line: 5]
< Func: main returns type void [mem: Global loc: 0 size: -2] [line: 1]
---
> .   .   Child: 0  Var: a of type int[mem: None loc: 0 size: 1] [line: 3]
> .   .   .   Child: 0  Var: c of type char[mem: None loc: 0 size: 1] [line: 6]
> .   .   .   Child: 0  Var: e of type bool[mem: None loc: 0 size: 1] [line: 10]
> .   Child: 1  Compound [mem: None loc: 0 size: 1] [line: 2]
> .   .   Child: 1  Compound [mem: None loc: 0 size: 1] [line: 5]
> Func: main returns type void[mem: None loc: 0 size: 1] [line: 1]
9,13c9,12
< Offset for end of global space: 0
< .   .   Sibling: 1  Compound [mem: None loc: 0 size: -6] [line: 9]
< .   .   Sibling: 1  Var: b of type int [mem: Local loc: -3 size: 1] [line: 3]
< .   .   .   Sibling: 1  Var: d of type char [mem: Local loc: -5 size: 1] [line: 6]
< .   .   .   Sibling: 1  Var: f of type bool [mem: Local loc: -5 size: 1] [line: 10]
---
> .   .   Sibling: 1  Compound [mem: None loc: 0 size: 1] [line: 9]
> .   .   Sibling: 1  Var: b of type int[mem: None loc: 0 size: 1] [line: 3]
> .   .   .   Sibling: 1  Var: d of type char[mem: None loc: 0 size: 1] [line: 6]
> .   .   .   Sibling: 1  Var: f of type bool[mem: None loc: 0 size: 1] [line: 10]
