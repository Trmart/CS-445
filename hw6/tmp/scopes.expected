.   .   Child: 0  Var: a of type int [mem: Local loc: -2 size: 1] [line: 3]
.   .   .   Child: 0  Var: c of type char [mem: Local loc: -4 size: 1] [line: 6]
.   .   .   Child: 0  Var: e of type bool [mem: Local loc: -4 size: 1] [line: 10]
.   Child: 1  Compound [mem: None loc: 0 size: -4] [line: 2]
.   .   Child: 1  Compound [mem: None loc: 0 size: -6] [line: 5]
Func: main returns type void [mem: Global loc: 0 size: -2] [line: 1]
Number of errors: 0
Number of warnings: 6
Offset for end of global space: 0
.   .   Sibling: 1  Compound [mem: None loc: 0 size: -6] [line: 9]
.   .   Sibling: 1  Var: b of type int [mem: Local loc: -3 size: 1] [line: 3]
.   .   .   Sibling: 1  Var: d of type char [mem: Local loc: -5 size: 1] [line: 6]
.   .   .   Sibling: 1  Var: f of type bool [mem: Local loc: -5 size: 1] [line: 10]
WARNING(10): The variable 'e' seems not to be used.
WARNING(10): The variable 'f' seems not to be used.
WARNING(3): The variable 'a' seems not to be used.
WARNING(3): The variable 'b' seems not to be used.
WARNING(6): The variable 'c' seems not to be used.
WARNING(6): The variable 'd' seems not to be used.
